components/analytics/team_analytics/team_analytics.tsx

```diff
@@ -3,28 +3,45 @@
 
 import PropTypes from 'prop-types';
 import React from 'react';
-import {FormattedDate, FormattedMessage} from 'react-intl';
-import {General} from 'mattermost-redux/constants';
+import { FormattedDate, FormattedMessage } from 'react-intl';
+import { General } from 'mattermost-redux/constants';
+import { Team, TeamStats } from 'mattermost-redux/types/teams';
+import { RelationOneToOne } from 'mattermost-redux/types/utilities';
 
 import LoadingScreen from 'components/loading_screen';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 
 import * as AdminActions from 'actions/admin_actions';
 import BrowserStore from 'stores/browser_store';
-import {StatTypes} from 'utils/constants';
+import { StatTypes } from 'utils/constants';
 import Banner from 'components/admin_console/banner';
 import LineChart from 'components/analytics/line_chart';
 import StatisticCount from 'components/analytics/statistic_count';
 import TableChart from 'components/analytics/table_chart';
 
-import {getMonthLong} from 'utils/i18n';
+import { getMonthLong } from 'utils/i18n';
 
-import {formatPostsPerDayData, formatUsersWithPostsPerDayData} from '../format';
+import { formatPostsPerDayData, formatUsersWithPostsPerDayData } from '../format';
 
 const LAST_ANALYTICS_TEAM = 'last_analytics_team';
 
-export default class TeamAnalytics extends React.PureComponent {
+//TODO find out where the types for these are initialized if you do end up using these
+type Props = {
+    actions: any;
+    teams: Team[];
+    stats: RelationOneToOne<Team, TeamStats>;
+    locale: string;
+}
+
+type State = {
+    team: Team | null;
+    recentlyActiveUsers: [];
+    newUsers: [];
+    id: string;
+}
+
+export default class TeamAnalytics extends React.PureComponent<Props, State> {  //? do I have to add the <P,S> parameters here?
     static propTypes = {
```

Comment:
`propTypes` should be removed altogether since TypeScript will handle the type-checking for us.